plugins/modules/packaging/os/pacman.py

```diff
@@ -405,6 +407,7 @@ def _install_packages_for_real(pacman_verb, pkglist):
             if rc != 0:
                 self.fail("Failed to install package(s)", cmd=cmd, stdout=stdout, stderr=stderr)
             self.add_exit_infos(stdout=stdout, stderr=stderr)
+            self._invalidate_database()
```

Comment:
Are you sure `_invalidate_database()` needs to be called from install, remove and upgrade?
My understanding is that `pacman --sync --list` dumps the list of packages available for install, this is should only change when `update_package_db()` is run.
Unless I'm missing something, I think you could leave out all invalidate calls except the one in `update_package_db`

Reply:
`pacman --sync --list` contains `[installed]` markers, so at least for install and remove it should be called. For upgrade the output also changes: for example after `pacman -Sy` it contained the entry `extra upower 0.99.17-1 [installed: 0.99.16-2]`, and after `pacman -Su` that entry changed to `extra upower 0.99.17-1 [installed]` for my system.